### PR TITLE
Bulk updates from initial

### PR DESCRIPTION
**Pull request by vladsavelyev**
Propagating recent changes in `initial`:
* Moving storage provider to cpg-utils
* Use TOML configuration, remove CLI
* Add sequencing type
* Refactor process substitution
* Creating ES-index SMDB analysis
* Filter samples by sequencing type
* Various fixes

### CODE REVIEW

File: cpg_pipes/pipeline/pipeline.py [hunk 1]
```diff
@@ -618,25 +620,34 @@ def _outputs_are_reusable(self, expected_out: ExpectedResultT) -> bool:
...
+                    _find_paths(el)
+            if isinstance(val, dict):
+                for el in val.values():
+                    _find_paths(el)
+
+        _find_paths(expected_out)
+
+        if paths and get_config()['workflow'].get('check_expected_outputs'):
+            # Checking that all paths in the expected output exist:
+            return all(exists(path) for path in paths)
```
Comment (lgruen) [inline cpg_pipes/pipeline/pipeline.py:643]: Not blocking, but just checking in terms of design: is the plan to move away from checking the existence of files and instead rely on metamist to determine the state of everything?
(@illusional FYI)

File: cpg_pipes/pipeline/pipeline.py [hunk 2]
```diff
@@ -768,104 +775,73 @@ class Pipeline:
...
     ):
-        self.storage_provider = storage_provider
+        self._stages = stages
+        self.run_id = get_config()['workflow'].get('run_id', timestamp())
+
+        analysis_dataset = get_config()['workflow']['dataset']
+        name = get_config()['workflow'].get('name') or name
+        description = get_config()['workflow'].get('description') or description
+        name = name or description or analysis_dataset
+        self.name = slugify(name)
```
Comment (lgruen) [inline cpg_pipes/pipeline/pipeline.py:789]: This is nice, as it'll make the job searchable in the UI.

File: cpg_pipes/pipeline/pipeline.py [hunk 3]
```diff
@@ -944,7 +916,9 @@ def set_stages(
         # implicit_stages_d: dict[str, Stage] = dict()  # If there are required
         # dependency stages that are not requested explicitly, we are putting them
         # into this dict as `skipped`.
+        depth = 0
         while True:  # Might need several rounds to resolve dependencies recursively.
```
Comment (lgruen) [inline cpg_pipes/pipeline/pipeline.py:920]: Should this include a check for cyclic dependencies?

**Review by lgruen — APPROVED** [review]
Very nice, thanks!

@illusional also had a few fixes recently (e.g. regarding process substitution and CRAM reference paths). Would be great to merge them into `main` soon as well, if it hasn't already happened.

File: cpg_pipes/pipeline/pipeline.py (hunk 1 above)
Comment (vladsavelyev) [reply cpg_pipes/pipeline/pipeline.py:643]: I didn't plan to make metamist a hard dependency, but I'm open to this suggestion!

File: cpg_pipes/pipeline/pipeline.py (hunk 3 above)
Comment (vladsavelyev) [reply cpg_pipes/pipeline/pipeline.py:920]: Not here. This code got has grown out of hand a bit and became too complicated. Essentially it's just topological sort here, and I have plans to switch to a more robust algorithm, but that's after I got everything in main.

File: cpg_pipes/pipeline/pipeline.py (hunk 1 above)
Comment (vladsavelyev) [reply cpg_pipes/pipeline/pipeline.py:643]: I'm more open to relying on metamist more, since now it's very stable and has a GUI

**Pull request merged**